Tasks/CondaClientInstallerV0/condaclientinstallermain.ts

```diff
@@ -0,0 +1,97 @@
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as rp from 'request-promise';
+import * as fs from 'fs'
+import * as util from 'azure-pipelines-tasks-packaging-common/util';
+import * as pkgLocationUtils from 'azure-pipelines-tasks-packaging-common/locationUtilities';
+import * as url from 'url';
+import * as constants from './constants';
+import { MinicondaRepo, AnacondaRepo, CondaEntry } from './condarepo';
+
+import { Platform, exec } from 'azure-pipelines-task-lib/task';
+// import { request } from 'http';
+
+async function main(): Promise<void> {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+    try {
+        let minicondaReleasesIndexUrl = "https://repo.anaconda.com/miniconda/";
+        let anacondaReleasesIndexUrl = "https://repo.anaconda.com/archive/";
+        
+        let distributionType = tl.getInput(constants.CondaInstallTaskInput.DistributionType);
+
+        const condaRepoPromise = distributionType === "Anaconda" ? AnacondaRepo.loadVersionsFromRepo(anacondaReleasesIndexUrl) : MinicondaRepo.loadVersionsFromRepo(minicondaReleasesIndexUrl);
+
+        let installLatest = tl.getBoolInput(constants.CondaInstallTaskInput.InstallLatest);
+        let pythonMajorVersion = tl.getInput(constants.CondaInstallTaskInput.PythonMajorVersion);
+        let pythonMinorVersion = tl.getInput(constants.CondaInstallTaskInput.PythonMinorVersion);
+        let condaVersion = tl.getInput(constants.CondaInstallTaskInput.CondaVersion);
+        let processorArchitecture = tl.getInput(constants.CondaInstallTaskInput.ProcessorArchitecture);
+        let installPath = tl.getInput(constants.CondaInstallTaskInput.InstallPath);
```

Comment:
Does the user need to decide where to install conda to?

Reply:
It's an optional parameter. There is a default value populated for that.